packages/material-ui-unstyled/src/SliderUnstyled/SliderUnstyled.js

```diff
@@ -626,9 +644,20 @@ const SliderUnstyled = React.forwardRef(function SliderUnstyled(props, ref) {
       {...other}
       className={clsx(utilityClasses.root, rootProps.className, className)}
     >
-      <Rail {...railProps} className={clsx(utilityClasses.rail, railProps.className)} />
+      <Rail
+        {...railProps}
+        {...(isComponent(Rail) && {
```

Comment:
When would `Rail` not be a component but valid for `<Rail />`? Do  we no longer transpile to React's JSX runtime?

Reply:
When it's a React primitive.

Reply:
Exactly we don't want to spread non DOM elements to primitives.

Reply:
> When it's a React primitive, e.g. 'span' by default.

That's a component. How do you determine from the element type that the props are not spread to host components? I don't think you can check that at runtime.

Reply:
Rail in this case is defined as:

```
const Rail = components.Rail || 'span';
```

It is behaving as the `component` prop on the other components that we have, it just stored under one `components` prop. I am checking for this value.

I've added tests for this in https://github.com/mui-org/material-ui/pull/22893/commits/c2ca7a697c516a0b5987d2384cd7e84ac3e65c0d

Reply:
Maybe we could rename the `isComponent` method? It's meant to answer: "can developers intercept the props". With a string primitive component => nop, with a custom component => yes

Reply:
The main problem is the naming here: `'span'` **is** a component. It's just implemented by `react-dom` instead of e.g. via function component. Maybe you mean `!isHostComponent(Rail)`? The current naming is confusing to me since a) we have React.isValidElementType for that and b) creating an element from something that isn't a component throws at runtime.

"host components" are also referred to as "built-in components" or "platform-specific components". "components" are synonymous with "element types". See https://reactjs.org/docs/implementation-notes.html#mounting-host-elements.

Reply:
`isHostComponent` and then negate it everywhere should probable be equivalent. I understand it's not introduced with this PR but I will stumble over this again the next time and others might also. Let's not proliferate inaccurate terminology.

We have precedent for "host component" in `elementAcceptingRef` and `elementTypeAcceptingRef` though these were introduced by myself.

Reply:
👍 for `isHostComponent`.

Reply:
Done 👍